examples/knative/python/kn-py-cbc-slack/deploy/handler.py

```diff
@@ -0,0 +1,98 @@
+    url = os.environ['SLACK_URL']
+    try:
+        slackWebhook = WebhookClient(url)
+    except:
+        return "Could not for Webhook to Slack"
+    app.logger.debug(f"Slack Webhook object is {slackWebhook}")
+
+    # Search CBC workload API for VM which event relates to
+    vmName = str(data['Vm']['Name'])
+    app.logger.debug(f"Virtual Machine name to search for is {vmName}")
+
+    try:
+        cbcComputeResourceQuery = workloadApi.select(ComputeResource).set_name([vmName])
+    except:
+        return "Could not form API query using the Virtual Server Name", 400
+    app.logger.debug(f"CBC compute resource query object is {cbcComputeResourceQuery}")
+
+    # Get VM objectc from CBC instance, if not immediately available wait and retry
+    tries = 5
+    while tries >=0:
+        try:
+            for vm in cbcComputeResourceQuery:
+                app.logger.debug(f"CBC compute resource object ID is " + vm.id)
+                if vm.id is None or tries == 0:
+                    app.logger.debug(f"Resource not found in CBS instance,  retrying")
+                    time.sleep(5)
+                    tries -= 1
+                else:
+                    app.logger.debug(f"Found VM in CBS instance")
+                    tries = 0
+        except:
+            return "Could not execute API query using the Virtual Server Name", 400
+    if vm.id is None:
+        app.logger.debug(f"VM not found in CBS instance")
+        return "", 400
+
+    # Query VM objet sensor install eligibility status, if not immediatly eligible wait and retry
+    tries = 5
+    while tries >=0:
+        try:
+            for vm in cbcComputeResourceQuery:
+                if vm.eligibility == 'ELIGIBLE':
+                    app.logger.debug(f"VM is eligible for sensor installation")
+                    try:
+                        workloadApi.select(ComputeResource,vm.id).install_sensor(os.environ['SENSOR_VER'],config_file=os.environ['CBC_CONFIG_INI'])
+                    except:
+                        return "Request to install CB sensor failed", 400
+                    app.logger.debug(f"Attempting installation of sensors on VM")
+                    try:
+                        slackResponse = slackWebhook.send(text=f":carbonblack: New VM - Carbon Black Installed :carbonblack: \n *CLOUDEVENT_ID*: \n {attrs['id']}\n\n Source:  {attrs['source']}\n Type:  {attrs['type']}\n *Subject*:  *{attrs['subject']}*\n Time:  {attrs['time']}\n\n *EVENT DATA*:\n key:  {data['Key']}\n user:  {data['UserName']}\n datacenter:  {data['Datacenter']['Name']}\n Host:  {data['Host']['Name']}\n VM:  {data['Vm']['Name']}\n\n Message: {data['FullFormattedMessage']}")
+                    except:
+                        return "Request to post Slack message failed", 400
+                    return "", 202
+                else:
+                    app.logger.debug(f"VM not eligible for sensor installation,  retrying " + vm.eligibility)
+                    time.sleep(30)
```

Comment:
nit: we should reduce this to say 10s because this is a blocking call and the cloudevent sender (broker) might not wait that long and consider your function to be dead.